### ISSUE TITLE
Selfie Recognition using DL

### ISSUE THREAD

**Issue by abhisheks008** [World-of-ML/DL-Simplified#242]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Selfie Recognition</br>
:red_circle: **Aim** : Create a DL model which will identify which one is selfie and which one is not.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/tapakah68/selfies-id-images-dataset</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by rafiya2003** [comment]
I would like to work on this, frankly I don't know much but I will learn and do. I am a newbie(first year UG student). I want to learn and grow

**Comment by abhisheks008** [comment]
Please do not to spam in every single issue. This might cause escalation as you are breaching Code of Conduct and Contribution Guidelines.

@rafiya2003

**Comment by stiwari-ds** [comment]
Full name : Siddhant Tiwari
GitHub Profile Link : github.com/stiwari-ds
Email ID : siddhant.tiwari.ds@gmail.com
Participant ID (if applicable): stiwari-ds (Quine)

Approach for this Project : 
1. Implement preprocessing methods to isolate faces and segment background using OpenCV
2. Train custom CNN models with processed vs. unprocessed images to compare results
3. Use pre-trained CNN architectures for feature-extraction as well as fine-tuning
4. Experiment with Vision Transformer models for treating images as a whole without any preprocessing

What is your participant role? (Mention the Open Source program): Individual Contributor, SSOC-2023

**Comment by abhisheks008** [comment]
Issue assigned to you @stiwari-ds

**Comment by atharv1707** [comment]
Full name : Atharv Pal
GitHub Profile Link : https://github.com/atharv1707
Email ID : atharvpal0799@gmail.com
Participant ID (if applicable): <I am new to Open Source, do tell me how to find it , I assume it's atharv1707>
Approach for this Project : 

Here is my suggested approach for this issue :

1. Exploratory Data Analysis (EDA): Understand the dataset's characteristics, including image distribution, resolution, and color composition.
2. Preprocessing: Standardize images for training by resizing, converting to appropriate formats, and normalizing pixel values.
3. Model Architecture Selection: Experiment with CNN architectures like VGG, ResNet, and Inception, exploring transfer learning for efficiency.
4. Model Training: Train models on a subset of the dataset, by leveraging data augmentation to prevent overfitting.
5. Evaluation: Assess model performance using accuracy, precision, recall, and F1-score metrics to identify the best algorithm.
6. Hyperparameter Tuning: Fine-tune model parameters such as learning rates and batch sizes for optimization.

What is your participant role? (Mention the Open Source program) Contributor, GSSoC'-2024.

**Comment by abhisheks008** [comment]
Well your approach looks good to me. You can start working on this issue @atharv1707. No need to worry about the participant ID, that's not applicable to you.

**Comment by atharv1707** [comment]
Hey! I tried for the past 3 hours working on this, and I have thoroughly gone through out the dataset concluded that the dataset is  not normalized, with selfie and id image ratio way too skewed. Also, I believe there is somewhere in between the images are mislabelled, which may in turn require manual labelling.

I request @abhisheks008  please unassign me from this issue. There are a few other projects I would like to contribute to. 

Thanks man!

**Comment by abhisheks008** [comment]
Cool @atharv1707

**Comment by Aarya31** [comment]
Full Name: Aarya Prajapat
Github Profile Link: https://github.com/Aarya31
Email Id - aaryaprajapat85@gmail.com
Participant ID - Aarya31

Approach for this Project:
Firstly, i would do data augmentation, standardize the images for training.
As it is image classification/selection problem, i will use ResNet, inception, DenseNet for efficiency and effectiveness.
Train the model and assess model performance using accuracy, precision,etc.

**Comment by abhisheks008** [comment]
Hi @Aarya31 issue assigned to you.
